### PR TITLE
Workflow methods

### LINKED ISSUE #77: include methods documentation in workflow version details
Make use of data describing the workflow.
This will require https://github.com/Duke-GCB/bespin-api/pull/83

### PR DESCRIPTION
**Pull request by johnbradley**
Adds workflow methods to the workflow versions detail screen.
Adds links so users can drill down into the workflow versions.
Sorts workflows by name.
Sorts workflow-versions by version number descending (current first).
Only shows the current version on the workflows list screen.
Fixes #77

### CODE REVIEW

**Comment by johnbradley** [comment]
That's strange. It cloned the repo into /tmp, checked out this brand and `npm install` seed to work fine for me?

**Comment by dleehr** [comment]
Tried that too.

```
git clone git@github.com:Duke-GCB/bespin-ui.git
cd bespin-ui/
git checkout workflow-methods 
npm install
ember s
```
The npm install succeeds but the markdown doesn't render and I get this in console

ember-markdown-render
ember-markdown-console

I see that my package-lock.json is changed, so commit that before doing anything else:  https://github.com/Duke-GCB/bespin-ui/tree/workflow-methods-package-lock

```
git add package-lock.json
git commit -m "package-lock after npm install"
```

Manually installing ember-cli-showdown fixes it, and updates package.lock.json again:

```
ember install ember-cli-showdown
git add package-lock.json
git commit -m "package-lock after ember install ember-cli-showdown"
npm install
ember s
```

ember-markdown-fixed

**Comment by dleehr** [comment]
Maybe my node version is too old. Or too new. ¯\_(ツ)_/¯

**Comment by dleehr** [comment]
I also updated npm in the middle of this, which likely threw a bunch of stuff into package.json.lock

File: app/components/workflows/workflow-detail.js [hunk 1]
```diff
@@ -2,4 +2,12 @@ import Ember from 'ember';
 
 export default Ember.Component.extend({
   workflow: null,
+  onlyShowCurrent: false,
+  sortedVersions: Ember.computed('onlyShowCurrent', 'workflow.versions', function() {
+    const versions = this.get('workflow.versions').sortBy('version').reverse();
```
Comment (dleehr) [inline app/components/workflows/workflow-detail.js]: I think this technically should compute on `'workflow.versions[]'`.

File: app/models/workflow-methods-document.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import DS from 'ember-data';
+import Ember from 'ember';
+
+export default DS.Model.extend({
+  workflowVersion: DS.belongsTo('workflow-version'),
+  content: DS.attr('string'),
+  contents: Ember.computed.alias('content'),
```
Comment (dleehr) [inline app/models/workflow-methods-document.js]: Can you add a comment here documenting why the duplication

File: app/templates/components/workflows/workflow-version-detail.hbs [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
-        <strong>{{title}}</strong>
+        <span class="lead">{{title}}</span>
     </div>
     <div class="panel-body">
-        <dl class="dl-horizontal">
-            <dt>Description</dt><dd class="workflow-version-details-description">{{workflowVersion.description}}</dd>
-            <dt>Created</dt><dd class="workflow-version-details-created">{{workflowVersion.created}}</dd>
-            <dt>URL</dt><dd class="workflow-version-details-url"><a href={{workflowVersion.url}}>{{workflowVersion.url}}</a></dd>
-        </dl>
+        {{markdown-to-html workflowMarkdown headerLevelStart=4 class="worklflow-version-detail-markdown"}}
```
Comment (dleehr) [inline app/templates/components/workflows/workflow-version-detail.hbs]: typo in "worklflow-version-detail-markdown"

File: tests/unit/models/workflow-methods-document-test.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  // Specify the other units that are required for this test.
+  needs: []
+});
+
+test('it exists', function(assert) {
+  let model = this.subject();
+  // let store = this.store();
+  assert.ok(!!model);
+});
+
```
Comment (dleehr) [inline tests/unit/models/workflow-methods-document-test.js]: Can the content/contents alias be tested here?

File: app/components/workflows/workflow-detail.js [hunk 5]
```diff
@@ -2,4 +2,12 @@ import Ember from 'ember';
...
+    if (this.get('onlyShowCurrent')) {
+      return versions.slice(0, 1);
```
Comment (dleehr) [inline app/components/workflows/workflow-detail.js]: FYI, Ember adds some convenient, observable properties to [Array](https://www.emberjs.com/api/ember/2.15/classes/Ember.Array/) like [firstObject](https://www.emberjs.com/api/ember/2.15/classes/Ember.Array/properties/firstObject?anchor=firstObject). Handy if like me, you can't remember how to use `slice()` and `splice()`

So you can write this line as versions.get('firstObject'), and sometimes even move to a computed property alias

**Pull request merged**